rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3124,6 +3124,18 @@ static rmw_ret_t rmw_send_response_request(
   }
 }
 
+static bool check_for_response_reader(dds_entity_t request_reader, dds_entity_t response_writer)
+{
+  dds_subscription_matched_status sm;
+  dds_publication_matched_status pm;
+  if (dds_get_subscription_matched_status(request_reader, &sm) < 0 ||
+    dds_get_publication_matched_status(response_writer, &pm) < 0)
+  {
+    return RMW_RET_ERROR;
```

Comment:
@eboasson 
```suggestion
    return false;
```
?

Reply:
Oops! Thanks! It’ll probably never get there (the function can’t fail if the reader/writer exist) but that’s no excuse for messing up the type so badly ...

Reply:
+1 to discriminating error conditions from entity absence.